server/src/controllers/company.ts

```diff
@@ -214,3 +222,151 @@ export async function deleteCompany(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  let user;
+  try {
+    user = await getUser(req.user?.email);
+  } catch (e) {
+    return next(e);
+  }
+  res.json(user.studentProfile?.companiesSeen?.length || 0);
+}
+
+/**
+ * Returns the internships of the previously seen companies by the user
+ * @param req
+ * @param res
+ * @param next
+ */
+export async function findInternshipsOfSeenCompanies(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  let user;
+  try {
+    user = await getUser(req.user?.email);
+  } catch (e) {
+    return next(e);
+  }
+  if (!user.studentProfile?.companiesSeen) res.json([]);
+  const internships = await collectInternships(user);
+
+  res.json(internships);
+}
+
+export async function collectInternships(user: IUser): Promise<any[]> {
+  let select = INTERNSHIP_FIELDS_VISIBLE_FOR_USER;
+  if (user.isAdmin) select += " " + INTERNSHIP_FIELDS_ADDITIONALLY_VISIBLE_FOR_ADMIN;
+  const projection = getProjection(select);
+  projection.company = { $first: "$company" };
+
+  const pipeline: unknown[] = [
+    {
+      $lookup: {
+        from: "companies",
+        localField: "company",
+        foreignField: "_id",
+        as: "company",
+      },
+    },
+    { $project: projection },
+    {
+      $match: {
+        "company._id": {
+          $in: user.studentProfile?.companiesSeen,
+        },
+      },
+    },
+  ];
+  const internships = await Internship.aggregate(pipeline);
+  return internships;
+}
+
+/**
+ * Returns amount of companies that fit certain search criteria eg. company.companyName
+ * and have not been seen yet by the user.
+ * @param req
+ * @param res
+ * @param next
+ */
+export async function findNewCompaniesAmount(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  let user;
+  try {
+    user = await getUserWithInternshipModule(req.user?.email);
+  } catch (e) {
+    return next(e);
+  }
+
+  // Create Options
+  const options = createInternshipQueryOptions(req.query);
+
+  if (!user.isAdmin && options.length !== 0) {
+    options["company.excludedFromSearch"] = false;
+    options.status = InternshipStatuses.PASSED;
+    const excludedInternships = user.studentProfile?.internship.internships || [];
+    if (excludedInternships.length > 0) {
+      options._id = {
+        $nin: excludedInternships,
+      };
+    }
+  }
+
+  const projection = getProjection(INTERNSHIP_FIELDS_VISIBLE_FOR_USER);
+  projection.company = { $first: "$company" };
+  //gets internships matching criteria but groups the ones in the same company
+  const pipeline: unknown[] = [
+    {
+      $lookup: {
+        from: "companies",
+        localField: "company",
+        foreignField: "_id",
+        as: "company",
+      },
+    },
+    {
+      $match: options,
+    },
+    {
+      $group: {
+        _id: "$company._id",
+        name: { $first: "$company.companyName" },
+        companyCount: { $sum: 1 },
+      },
+    },
```

Comment:
Wenn hier noch ein Count-Step in der Pipeline dazukommt, muss das auch nicht mehr außerhalb der Pipeline gemacht werden:

```ts
{
  $count: "count",
},
```

Und dann unten einfach nur noch `res.json(companiesWithInternships[0].count);` im else-Block